### ISSUE TITLE
Hard Limit at 193 Frames

### ISSUE THREAD

**Issue by pftq** [SkyworkAI/SkyReels-V1#63]
The readme mentions testing on at 200+ frames, but the model seems to reset at  the193rd frame no matter how much VRAM is available (tested on H100, etc).

**Comment by pftq** [comment]
I think I found the cause of the 192-frame limit. It's in the VAE, specifically autoencoder_kl_hunyuan_video.py. The original code runs into an out-of-bounds at the 25th tile (193rd frame at 8 frames per tile).

Find:

for i in range(0, num_frames, tile_latent_stride_num_frames):
    tile = z[:, :, i : i + tile_latent_min_num_frames + 1, :, :]
Replace:

for i in range(0, num_frames, tile_latent_stride_num_frames):
    end_idx = min(i + tile_latent_min_num_frames + 1, num_frames)
    tile = z[:, :, i : end_idx, :, :]
    if i > 0:
        tile = torch.cat([z[:, :, i - 1 : i, :, :], tile], dim=2)  # Add previous frame for continuity

Is this something that SkyReels can fix on their end or would the Hunyuan team have to push this change?

**Comment by pftq** [comment]
I've added a fix to the 192-frame-limit bug by applying the Riflex extrapolation technique by thu-ml (credit Kijai for using it in ComfyUI and making me aware of it).  This is a pretty solid workaround until there's a true fix for why the video turns to static noise on frame 193 and resets.
https://github.com/SkyworkAI/SkyReels-V1/pull/83/files#diff-23418e8cc57144ed095f778f599e57792d2c651852c1fe66419afaa2cf2cf878
